src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -348,6 +350,29 @@ abstract class AbstractTreeView extends Disposable implements ITreeView {
 		this._onDidChangeDescription.fire(this._description);
 	}
 
+	private _badge: IViewBadge | undefined;
+	private _badgeActivity: IDisposable | undefined;
+	get badge(): IViewBadge | undefined {
+		return this._badge;
+	}
+
+	set badge(badge: IViewBadge | undefined) {
+
```

Comment:
Before updating `this._badgeActivity` we should check to see if the properties of `badge` differ from `this._badge`.

Reply:
Done.